nixos/modules/services/networking/sozu.nix

```diff
@@ -0,0 +1,304 @@
+            default = "rustls";
+            description = ''
+              Defines how the TLS protocol will be handled by sozu. Possible values
+              are "openssl" or "rustls". The "openssl" option will only work if sozu
+              was built with the "use-openssl" feature.
+            '';
+          };
+
+          front_timeout = mkOption {
+            type = types.int;
+            default = 60;
+            description = ''
+              Maximum time of inactivity for a front socket, in seconds.
+            '';
+          };
+
+          zombie_check_interval = mkOption {
+            type = types.int;
+            default = 1800;
+            description = ''
+              Duration between zombie checks, in seconds.
+              In case of bugs in sozu's event loop and protocol implementations,
+              some client sessions could be stuck, not receiving any more event,
+              and consuming resources. Sozu verifies regularly if there are such
+              zombie sessions, logs their state and removes them.
+            '';
+          };
+
+          activate_listeners = mkOption {
+            type = types.bool;
+            default = true;
+            example = false;
+            description = ''
+              By default, all listeners start a TCP listen socket on startup.
+              If set to false, this option will prevent them from listening. You can then add
+              the complete configuration, and send an ActivateListener message afterwards.
+            '';
+          };
+
+        };
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra configuration for <link xlink:href="https://github.com/sozu-proxy/sozu/blob/master/doc/configure.md#metrics">metrics</link>, <link xlink:href="https://github.com/sozu-proxy/sozu/blob/master/doc/configure.md#listeners">listeners</link>,
+        and <link xlink:href="https://github.com/sozu-proxy/sozu/blob/master/doc/configure.md#applications">applications</link>.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [
+      sozu
+    ];
+
+    systemd.services.sozu = {
+      description = "Sozu - A HTTP reverse proxy, configurable at runtime, fast and safe, built in Rust.";
+      after = [ "network.target" ];
+      wants = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        # Create the proper run directory
+        mkdir -p /run/sozu
+        chown -R sozu /run/sozu
+      '';
+
+      serviceConfig = {
+        PIDFile = "/run/sozu/sozu.pid";
+        ExecStart = "${cfg.package}/bin/sozu start --config ${configFile}";
+        Restart = "on-failure";
+      };
+
+    };
+
+    users.users.sozu = {
+      description = "Sozu Daemon User";
+      isSystemUser = true;
+    };
+
+  };
+
+  meta = with lib; {
+    maintainers = with maintainer; [ netcrns ];
```

Comment:
Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME``.

Reply:
> https://github.com/NixOS/nixpkgs/pull/120010#issuecomment-827008692
I've also removed myself from the maintainers-list, in favour of doing that [here](https://github.com/NixOS/nixpkgs/pull/120754), seeing as that PR will probably be merged before this one.

> https://github.com/NixOS/nixpkgs/pull/120754#issuecomment-826974445
Seeing as this PR should presumably be merged before the aforementioned one (we're waiting for sozu v0.12.0 to be released), I'll remove the commit from that and move it here. Would that be alright?